### ISSUE TITLE
NVDA fails to identify alertdialog role in Chrome browser

### ISSUE THREAD

**Issue by sidnc86** [nvaccess/nvda#9368]
### Steps to reproduce:

1. Open Chrome
2. Browse to [ARIA 1.1 alertdialog example on W3C website](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/alertdialog.html)
3. There is a message compose form which can be saved to browser's local storage. Instead of clicking the Save button, click the Discard button.
4. A confirmation alertdialog box appears. 

### Actual behavior:

Only the focused button caption is read out on open of the alertdialog.

### Expected behavior:

As soon as the alertdialog modal is opened and button is focused, NVDA should read "Confirm dialog. Are you sure you want to delete 44 words...?" followed by focused button caption.

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

2018.4.1

#### Windows version:

Windows 10 Home Single Language

#### Name and version of other software in use when reproducing the issue:

Chrome browser version 72.0.3626.121 

#### Other information about your system:

I5 Processer on Lenovo Z50-70 model

#### Does the issue still occur after restarting your PC?

Yes

#### Have you tried any other versions of NVDA?

No

**Comment by sidnc86** [comment]
Also here is the NVDA log file that I forgot to attach when I posted the issue.
[nvda.log](https://github.com/nvaccess/nvda/files/2956689/nvda.log)

**Comment by ehollig** [comment]
This is still the case in the latest alpha. Sounds like another role that NVDA does not fully support. CC @leonardder

**Comment by bramd** [comment]
I tested with this data URL:

`data:text/html,<div role="alertdialog" aria-label="Warning" aria-describedby="foo"><span id="foo">Very serious warning!</span></div>`

NVDA adds the Dialog overlay class as expected, but it also assigns a role of alert. I'm not sure where this role is coming from. Can anyone give me some pointers on where the alert role may be set?

NVDA dev info in Chrome:

```
Developer info for navigator object:
name: 'Warning'
role: ROLE_ALERT
roleText: None
states: STATE_FOCUSABLE, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_DialogDialogIa2WebIAccessible object at 0x08BD4FF0>
Python class mro: (<class 'NVDAObjects.Dynamic_DialogDialogIa2WebIAccessible'>, <class 'NVDAObjects.behaviors.Dialog'>, <class 'NVDAObjects.IAccessible.ia2Web.Ia2Web'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'object'>)
description: 'Very serious warning!'
location: RectLTWH(left=8, top=79, width=1264, height=18)
value: None
appModule: <'appModuleHandler' (appName 'chrome', process ID 10896) at address 5c29110>
appModule.productName: 'Google Chrome'
appModule.productVersion: '78.0.3904.108'
TextInfo: <class 'NVDAObjects.IAccessible.IA2TextTextInfo'>
windowHandle: 2492516
windowClassName: 'Chrome_RenderWidgetHostHWND'
windowControlID: 1258208
windowStyle: 1445986304
extendedWindowStyle: 32
windowThreadID: 18116
windowText: 'Chrome Legacy Window'
displayText: ''
IAccessibleObject: <POINTER(IAccessible2) ptr=0x58c267c at 8cde030>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=2492516, objectID=-4, childID=-91569
IAccessible accName: 'Warning'
IAccessible accRole: ROLE_SYSTEM_ALERT
IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048580)
IAccessible accDescription: 'Very serious warning!'
IAccessible accValue: ''
IAccessible2 windowHandle: 2492516
IAccessible2 uniqueID: -91569
IAccessible2 role: ROLE_SYSTEM_ALERT
IAccessible2 states: IA2_STATE_OPAQUE (1024)
IAccessible2 attributes: 'display:block;tag:div;xml-roles:alertdialog;explicit-name:true;'
```

**Comment by Adriani90** [comment]
cc: @feerrenrut sorry for CCing you on two similar issues, this one contains a more valuable discussion.

**Comment by aleventhal** [comment]
It's actually a regression from a couple of years ago. I've filed https://bugs.chromium.org/p/chromium/issues/detail?id=1071838  and I believe that using xml-roles:alertdialog along with ROLE_SYSTEM_DIALOG but still get the benefit of JAWS dialog commands, and automatically speaking the alertdialog even if it doesn't receive focus. More info in the Chrome bug.

I should be able to get a fix into Chrome 84. I probably won't want to merge it to Chrome 83, but you never know.

**Comment by Adriani90** [comment]
#8620 might be related

**Comment by Adriani90** [comment]
The Chrome bug is fixed and this works as expected now with Chrome 111 and NVDA 2023.1 RC1. Closing as works for me.
Thanks @aleventhal for the great work you've done over the last years to fix so many accessibility bugs in Chromium. Really appreciated.
